### ISSUE TITLE
Request headers are dropped when compressed `Content-Encoding` is used

### ISSUE THREAD

**Issue by RomanIakovlev** [playframework/playframework#7911]
### Expected Behavior
Regardless of `Content-Encoding`, the unrelated headers like `Authorization` or custom headers should be preserved. Particularly, when sending a request like:

```bash
curl -X POST http://localhost:9000 \
-H 'Content-Encoding: gzip' \
-H 'Authorization: Bearer 123' \
-H 'X-Custom-Header: 123' -vvv
```
Then `X-Custom-Header` and `Authorization` should be available in the `play.api.mvc.RequestHeader.headers` collection.

### Actual Behavior
However, the actually available headers in the `play.api.mvc.RequestHeader.headers` collection are the following: 
`List((Transfer-Encoding,chunked), (Content-Type,none/none), (Content-Encoding,gzip))`. 
As you can see, `X-Custom-Header` and `Authorization` headers are missing.

### Reproducible Test Case

The project (a basic Play app) helping to illustrate this issue can be found at
https://github.com/RomanIakovlev/play-headers-issue. Just run it in the dev mode via sbt, and execute the specified curl command to see the result.

**Comment by RomanIakovlev** [comment]
I've just tested the same project with Netty as a backend, and then this issue is not reproducible. It seems this problem comes either from Akka Http, or from its integration with Play.

**Comment by richdougherty** [comment]
Thanks for the report. I'll take a look at this one.

**Comment by richdougherty** [comment]
Verified in an integration test. Headers are present at the first level of Akka parsing, but are missing by the time Play receives Akka's request object. I'll debug a bit more to find when/why the headers disappear. This will probably require a patch to Akka HTTP.

**Comment by richdougherty** [comment]
I've submitted a fix for this: #7930.

**Comment by gmethvin** [comment]
The fix for this will be in 2.6.7 and is in the latest 2.6.x [nightly snapshot](https://playframework.com/documentation/2.6.x/Repositories#Accessing-nightly-snapshots), version [2.6.7-2017-10-18-e1366be-SNAPSHOT](https://oss.sonatype.org/content/repositories/snapshots/com/typesafe/play/sbt-plugin_2.12_1.0/2.6.7-2017-10-18-e1366be-SNAPSHOT/).
